apps/workplace/src/tests/book/new-desk-flow.component.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { FormGroup } from "@angular/forms";
+import { createRoutingFactory, SpectatorRouting } from "@ngneat/spectator/jest"
+import { BookingFormService } from "@placeos/bookings";
+import { OrganisationService } from "@placeos/organisation";
+import { MockComponent } from "ng-mocks";
+import { BehaviorSubject, of } from "rxjs";
+import { NewDeskFlowComponent } from "../../app/book/new-desk-flow.component";
+import { NewDeskFlowFormComponent } from "../../app/book/new-desk-flow/new-desk-flow-form.component";
+import { NewDeskFlowSuccessComponent } from "../../app/book/new-desk-flow/new-desk-flow-success.component";
+
+
+describe('NewDeskFlowComponent', () => {
+    let spectator: SpectatorRouting<NewDeskFlowComponent>;
+    const createComponent = createRoutingFactory({
+        component: NewDeskFlowComponent,
+        providers: [
+            {
+                provide: BookingFormService,
+                useValue: {
+                    loadForm: jest.fn(),
+                    newForm: jest.fn(),
+                    setView: jest.fn(),
+                    form: new FormGroup({}),
+                    view:'',
+                    last_success: null
+                }
+            },
+            {
+                provide: OrganisationService,
+                useValue: {
+                    initialised: new BehaviorSubject(true)
+                }
+            }
+        ],
+        declarations: [
+            MockComponent(NewDeskFlowFormComponent),
+            MockComponent(NewDeskFlowSuccessComponent)
+        ]
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+        const book_service:any = spectator.inject(BookingFormService);
+        book_service.setView.mockImplementation(e => {
+            book_service.view = e;
+            spectator.detectChanges();
+        });
+        book_service.setView('form');
+    });
+
+    it('should create component', () =>
+    expect(spectator.component).toBeTruthy());
+
+    it('should show form view by default', () => {
+        expect(spectator.query('new-desk-flow-form')).toExist();
```

Comment:
Note that you don't need to do `spectator.query` for `.toExist` checks.
`expect('new-desk-flow-form').toExist();` works the same and is less verbose.

Reply:
Fixed accordingly